x-pack/plugins/ml/public/components/upgrade/upgrade_warning.js

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+import React from 'react';
+
+import {
+  EuiCallOut,
+  EuiSpacer,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+
+let upgradeInProgress = false;
+
+export function setUpgradeInProgress(show) {
```

Comment:
The code is certainly fine as is, I was just thinking if this could be part of `services/job_service.js`. This component would then import `job_service` and call a `getUpgradeInProgress()` in its render function, `get_privileges.js` further down would call `jobService.setUpgradeInProgress()`.